components/containers/core/mojom/containers.mojom

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module containers.mojom;
+
+// Represents a container that holds a separate browsing context/identity.
+// Containers allow users to maintain different identities within the same
+// browser profile, keeping cookies, storage, and other site data separate
+// between containers.
+struct Container {
+  // Unique identifier for the container.
+  string id;
+  // User-friendly name for the container.
+  string name;
+};
+
+// Interface for settings UI -> browser communication. This interface allows the
+// settings UI to perform CRUD operations on containers, with all operations
+// being handled in the browser process.
+interface ContainersSettingsHandler {
+  // Establishes a connection with the UI for browser -> UI notifications. This
+  // creates a 1-to-1 connection between the handler and the settings UI.
+  BindUI(pending_remote<ContainersSettingsUI> ui);
+
+  // Retrieves the current list of containers. Returns: Array of all configured
+  // containers
+  GetContainers() => (array<Container> containers);
+
+  // Creates a new container or updates an existing one. A new container must
+  // have an empty ID, and an existing container must have a non-empty ID.
+  AddOrUpdateContainer(Container container);
+
+  // Removes a container by its ID. Returns async response to block UI until
+  // completion, because the data removal might take some time.
+  RemoveContainer(string id) => ();
```

Comment:
I think this can potentially fail so it should also have an error callback? Not sure how we're cleaning up, but `AsyncObliterateStoragePartition` can fail

Reply:
I agree if this function is in this PR then we should have the version of the function that we agree will work best for the intended implementation. If we're reviewing a mojom interface function it should be as close to the intended version as possible for the feature being iteratively implemented. The implementation can come iteratively - the callback doesn't need to cover every scenario yet.

Reply:
Discussed this with @petemill and he agrees that the mojo interface should not be done iteratively unless we make changes to the UX. Things like error handling can easily be missed and also making changes later requires changes to the UI for other platforms. It's ok not to implement the error handling in this PR, but there should be an error handling mechanism in the interface